src/test/java/org/araymond/joal/core/ttorrent/client/announcer/tracker/TrackerClientUriProviderTest.java

```diff
@@ -44,6 +44,7 @@ public void shouldMoveToNext() throws NoMoreUriAvailableException {
     @Test
     public void shouldDeleteAndMoveToNext() throws NoMoreUriAvailableException {
         final TrackerClientUriProvider provider = createOne("http://localhost", "https://localhost", "http://127.0.0.1", "https://127.0.0.1");
+        provider.moveToNext();
```

Comment:
Why are thos two addition neede (line 47 and 57) ?
Did one of your hcange broke a test ?

Reply:
Yes, this change:
https://github.com/anthonyraymond/joal/pull/172/commits/cd97ef1c21ae0d724607ee9e8002f98eb017ea6c#diff-0e12b048859183dfe96b55aaa1ecb533436ac94f6a43725a34539a68845b7b10L40-R39

If you'd prefer, we could call `this.get()` in the constructor to avoid this odd state.

Reply:
I see.

Then `TrackerClient` is passed a *non-provisioned* `TrackerClientUriProvider` in `Announcer#buildTrackerClient()` and possibily on some other place, wich would result in `NoMoreUriAvailableException` at start i guess.

I'l prefer your idea to have the init in the constructor, it could be even better if `TrackerClientUriProvider` had :
- private constructor 
- public factory method for that handle the first `moveToNext()` and fails if empty.